### PR TITLE
Enable Checkstyle for test sources

### PR DESCRIPTION
**Pull request by basil**
I kept seeing new unused imports crop up, which I thought I had fixed already by enabling the `UnusedImports` Checkstyle check. Then I realized that Checkstyle was not being run against test sources. This PR enables Checkstyle for test sources and cleans up any existing violations. The motivation is to keep code cleanups from regressing in the future.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -453,6 +453,7 @@ THE SOFTWARE.
           <configuration>
             <consoleOutput>true</consoleOutput>
             <failsOnError>true</failsOnError>
+            <includeTestSourceDirectory>true</includeTestSourceDirectory>
```
Comment (basil) [inline pom.xml:456]: The main part of the change, where we enable Checkstyle for test sources.

File: pom.xml [hunk 2]
```diff
@@ -473,7 +474,6 @@ THE SOFTWARE.
                   <module name="CovariantEquals" />
                   <module name="EqualsHashCode" />
                   <module name="OneStatementPerLine" />
-                  <module name="PackageDeclaration" />
```
Comment (basil) [inline pom.xml:476]: There weren't any violations of this rule in main sources, but there were some violations in test sources. Looking at the rule, it didn't seem to provide much value in the first place, so I chose to disable it for now. We can always enable it again in the future if we feel that the value it provides is worth fixing the violations in test sources.

File: test/src/test/java/hudson/model/RunTest.java [hunk 3]
```diff
@@ -219,7 +219,7 @@ public FullNameChangingProject newInstance(ItemGroup parent, String name) {
         }
     }
     
-    private static final class Mgr extends ArtifactManager {
+    public static final class Mgr extends ArtifactManager {
```
Comment (basil) [inline test/src/test/java/hudson/model/RunTest.java:222]: If we didn't make this `public`, then the `public` keyword for the `Factory` constructor on line 233 would be redundant, forcing us to remove it to satisfy Checkstyle's `RedundantImport` check, which would then break a different check which demands that anything annotated with `@DataBoundConstructor` be `public`. Looking at how other test classes deal with this problem, I see that most test code that uses `@DataBoundConstructor` is nested within a `public` parent class, so I opted to follow the same strategy here.

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by daniel-beck** [comment]
Could we please not overdo it with restrictions on test code? While I'd like to think I wouldn't do it, I would probably stop writing tests and call it good enough if this stuff starts getting on my nerves for no real reason.

**Comment by basil** [comment]
There _is_ a real reason for these checks (provided in the "Rationale:" section of each check's description), and enforcing them is not overdoing anything. Most of the existing violations were introduced by Kohsuke and others 7-15 years ago, and I am fixing all of those in this PR so you do not have to worry about them. The amount of "new" (2-3 years old) code being touched in this PR is minimal, so people should not be tripping over these checks regularly. If they are, we can always add an exclusion for whatever specific check(s) people find annoying rather than disabling all checks on all tests.

**Comment by daniel-beck** [comment]
> There is a real reason for these checks (provided in the "Rationale:" section of each check's description)

I just checked https://checkstyle.sourceforge.io/config_modifier.html#RedundantModifier and all it says is

> Public modifier for constructors in non-public non-protected classes is always obsolete

From my POV, that's not a reason. I know it's not necessary. It doesn't explain why we need to enforce that nobody does it.

This causes builds to fail, either during development, or while building a PR. What can possibly be the benefit of this that outweighs the drawbacks of wasting people's time, sometimes repeatedly?

> Most of the existing violations were introduced by Kohsuke and others 7-15 years ago

Correlation to the greatest development activity by lines changed/tests written from scratch?

**Comment by basil** [comment]
The point is that redundant and inconsistent constructs inhibit readability and that catching these problems early through automation is more efficient and reliable than catching them during code review or after the fact. For example, SECURITY-1452 unwittingly regressed James' work on JSR 305 annotations in `VirtualFileTest`; this is exactly the kind of infrequent mistake that is best caught by automation rather than code review. Naturally some checks have more value than others, but I am not going to debate the relative merits of specific checks because I have already offered to disable enforcement of any checks that people find annoying. The intention is for these checks to find violations rarely enough so as not to cause annoyance.

**Pull request merged**